src/MongoDB.Driver.Core/Core/Authentication/Sspi/SspiSecurityContext.cs

```diff
@@ -128,11 +122,13 @@ public void DecryptMessage(int messageLength, byte[] encryptedBytes, out byte[]
                     descriptor.Free();
                 }
             }
+
+            return decryptedBytes;
         }
 
-        public void EncryptMessage(byte[] inBytes, out byte[] outBytes)
+        public byte[] EncryptMessage(byte[] plainTextBytes)
```

Comment:
Is it necessary to change this from an out param to a return value?  Is it a matter of preference?

Reply:
The Win32 API makes heavy use of `out` parameters because methods indicate success/failure through the returned HRESULT. (GSSAPI does something similar in returning `majorCode` and `minorCode`.) .NET prefers exceptions for this purpose.

While it is somewhat of a personal preference to refactor from void/out to a return value, it has the benefit of making the code more comprehensible. For example, I was able to quickly spot that the return value of `DecryptMessage` was never used whereas when it was an out parameter, it wasn't immediately obvious.